### PR TITLE
Socket-activated systemd service for non-root use of usbsdmux

### PR DESCRIPTION
**Pull request by SmithChart**
These commits introduce a socket-activated systemd service. This service acts as a proxy for commands to the USB-SD-Mux. Using the proxy enables non-root users to control the USB-SD-Mux from their context.

A socket-activated service is used to reduce load on systems where the USB-SD-Mux is only used sporadically.

The command line interface of the `usbsdmux`tool is left unchanged as far as possible:

* Using it in the old syntax still enables a user to use it with root-privileges.
* Adding a `-c` enables client-mode and used the default socket-location.
* Adding a `-s /path/to/socket` enables client-mode with the specified socket.

Communication over the socket is done JSON-encoded.

### CODE REVIEW

File: usbsdmux/service.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                 written into the 'error'-field of the answer.
+    err_text -- Free Text that will be passed to the client along with the
+                error state.
+    """
+    answer = dict()
+    answer["error"] = had_error
+    answer["text"] =  err_text
+    return json.dumps(answer)
+
+def parse(raw_string):
```
Comment (jluebbe) [inline usbsdmux/service.py]: This executes the command as well as parse it. Maybe rename it to something else?

File: usbsdmux/__main__.py [hunk 2]
```diff
@@ -13,20 +15,46 @@ def main():
         help="mode to switch to",
         choices=["dut", "host", "off", "client"],
         type=str.lower)
+    parser.add_argument(
+        "-c",
+        "--client",
+        help="Run in client mode with socket /tmp/sdmux.sock",
+        action="store_true",
+        default=False)
```
Comment (jluebbe) [inline usbsdmux/__main__.py]: Maybe we could default to client mode for non-root users?

File: contrib/systemd/usbsdmux.service [hunk 3]
```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=USB-SD-Mux Proxy-Service
+After=network.target usbsdmux.socket
+Requires=usbsdmux.socket
+
+[Service]
+Type=simple
+ExecStart=/path/to/usb-sd-mux/.venv/bin/usbsdmux-service
+TimeoutStopSec=5
```
Comment (jluebbe) [inline contrib/systemd/usbsdmux.service]: Why do we need to reduce TimeoutStopSec?

Comment (SmithChart) [reply contrib/systemd/usbsdmux.service]: You are right. We can use the system default for TimeoutStopSec here.
I'll remove the TimeoutStopSec parameter.

File: usbsdmux/__main__.py (hunk 2 above)
Comment (SmithChart) [reply usbsdmux/__main__.py]: I added this feature in 791e514aff389d04d381862c4cdad3fa1df4234d. usbsdmux now tries to guess the best mode for the current user. Additionally a new switch is introduced to force usbsdmux into direkt mode.

File: usbsdmux/service.py (hunk 1 above)
Comment (SmithChart) [reply usbsdmux/service.py]: You are right. The name is misleading.

Renamed:
* :code:`parse()` to :code:`process_request()`
* :code:`answer()` to :code:`create_answer()`

**Comment by SmithChart** [comment]
@jluebbe: I had a look at your comments and fixed the issues. Could you have another look at it?

**Pull request merged**